### PR TITLE
fix: DEV-3743: Fixing aggregation for annotation results

### PR DESCRIPTION
**Pull request by wesleylima**

### CODE REVIEW

**Review by makseq — CHANGES_REQUESTED** [review]

**Comment by makseq** [comment]
Please add a test for this case, when anno result is empty

File: label_studio/data_manager/managers.py [hunk 1]
```diff
@@ -492,7 +492,7 @@ def annotate_annotations_results(queryset):
         return queryset.annotate(annotations_results=Coalesce(
             GroupConcat("annotations__result"), Value(''), output_field=models.CharField()))
     else:
-        return queryset.annotate(annotations_results=ArrayAgg("annotations__result", distinct=True))
+        return queryset.annotate(annotations_results=ArrayAgg("annotations__result", distinct=True, default=Value([])))
```
[the remark this answers is not in the document]
Comment (wesleylima) [reply label_studio/data_manager/managers.py]: This worked properly in sqlite before. Just re-confirmed now.

File: label_studio/data_manager/aggregates.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+from django.db.models.lookups import Exact
+from django.db.models import Func, Value
+from django.contrib.postgres import lookups
+from django.contrib.postgres.aggregates import ArrayAgg as OriginalArrayAgg
+from django.contrib.postgres.fields import ArrayField
+from django.contrib.postgres.fields.array import ArrayLenTransform as OriginalArrayLenTransform, ArrayInLookup as OriginalArrayInLookup
+
+
+class ArrayRHSMixin:
```
Comment (makseq) [inline label_studio/data_manager/aggregates.py:16]: I would solve it using another `if` statement here: https://github.com/heartexlabs/label-studio/blob/develop/label_studio/data_manager/managers.py#L195

Will it correctly work it with `[]` (empty list), when annotation.result = [] (instead of annotation.result = None) ?

File: label_studio/tests/tasks_api.tavern.yml [hunk 3]
```diff
@@ -24,4 +24,161 @@ stages:
     method: GET
     url: '{django_live_url}/api/tasks/{task_pk}?project={project_pk}'
   response:
-    status_code: 200
\ No newline at end of file
+    status_code: 200
+
```
Comment (makseq) [inline label_studio/tests/tasks_api.tavern.yml:28]: For data manager filters we have a special folder:
image

**Review by makseq — CHANGES_REQUESTED** [review]

**Comment by makseq** [comment]
@wesleylima Are you sure you reverted this branch to the correct commit? Please check this comment: https://github.com/heartexlabs/label-studio/pull/3297/files#r1052506144

**Review by makseq — CHANGES_REQUESTED** [review]
Are you sure you reverted this branch to the correct commit? 
https://github.com/heartexlabs/label-studio/pull/3297/files#r1052506144

File: label_studio/data_manager/aggregates.py (hunk 2 above)
Comment (wesleylima) [reply label_studio/data_manager/aggregates.py]: I don't want to make too many changes to this file. It's just a "fast forward" of aggregates.py to a newer django version without requiring the latest django version. I think it completely addresses the bug we're trying to solve as is.

Comment (makseq) [reply label_studio/data_manager/aggregates.py]: I still have two concerns here:
1. performance, how will it be affected? 
2. from user perspective `annotation.result = []` is empty annotation (you can try to get this by clicking submit without any region creation). However, this approach will process [] as not empty annotation.

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio/data_manager/managers.py [hunk 4]
```diff
@@ -195,7 +194,13 @@ def add_result_filter(field_name, _filter, filter_expressions, project):
     elif _filter.operator == Operator.NOT_CONTAINS:
         filter_expressions.append(~Q(subquery))
         return 'continue'
-
+    elif _filter.operator == Operator.EMPTY and field_name == 'annotations_results':
+        if _filter.value == 'true':
```
Comment (makseq) [inline label_studio/data_manager/managers.py]: Let's edit this a little: 
```  
cast_value(_filter)  # it works inplace
if _filter.value: 
  ...
```

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio/data_manager/managers.py (hunk 4 above)
Comment (wesleylima) [reply label_studio/data_manager/managers.py]: yes this is cleaner. Unfortunately the UI is sending us the "annotation_results is empty" filter typed as a String so `cast_value` won't cast it into a Boolean
```
{
  "filter": "filter:tasks:annotations_results",
  "operator": "empty",
  "value": "true",
  "type": "String"
}
```

**Review by makseq — APPROVED** [review]

**Review by makseq — APPROVED** [review]
